internal/controller/ingress/httpsedge_controller.go

```diff
@@ -1079,11 +1092,30 @@ func (u *edgeRouteModuleUpdater) setEdgeRoutePolicy(ctx context.Context, route *
 		return client.Delete(ctx, edgeRouteItem(route))
 	}
 
-	log.Info("Updating Policy module")
-	_, err := client.Replace(ctx, &ngrokapi.EdgeRoutePolicyRawReplace{
+	module := ngrok.EndpointTrafficPolicy{}
+
+	if util.IsLegacyPolicy(trafficPolicy) {
+		u.recorder.Eventf(u.edge, v1.EventTypeWarning, "UpdateWarning", "Traffic Policy is using legacy directions: ['inbound', 'outbound']. Update to new phases: ['on_tcp_connect', 'on_http_request', 'on_http_response']")
```

Comment:
Nit: The 3rd field of k8s events is a grouping key that can match events together and update their messages. I recommend using `Updated` for both success and failure events related to updating. Similar for `Created`, etc...

Reply:
Talked to Stacks about this. We are going to use `PolicyDeprecation` as the event reason here. That way, we can have users filter on that in their events when we actually need them to start fixing this stuff. It also aligns with how k8s does their event reasons.